----------

Haruo Suzuki (haruo[at]g-language[dot]org)  
Last Update: 2017-08-17

----------

# Codon usage

----------

## References

----------

### Amino acid usage

https://www.ncbi.nlm.nih.gov/pubmed/8065933
Nucleic Acids Res. 1994 Aug 11;22(15):3174-80.
Hydrophobicity, expressivity and aromaticity are the major trends of amino-acid usage in 999 Escherichia coli chromosome-encoded genes.
Lobry JR1, Gautier C.

Googled: "amino acid usage" mutational bias

https://www.ncbi.nlm.nih.gov/pubmed/16936139
Mol Biol Evol. 2006 Dec;23(12):2303-15. Epub 2006 Aug 25.
The evolution of biased codon and amino acid usage in nematode genomes.
Cutter AD1, Wasmuth JD, Blaxter ML.

https://www.ncbi.nlm.nih.gov/pubmed/14672975
Genome Res. 2004 Jan;14(1):44-53. Epub 2003 Dec 12.
Mutational and selective pressures on codon and amino acid usage in Buchnera, endosymbiotic bacteria of aphids.
Rispe C1, Delmotte F, van Ham RC, Moya A.

https://www.ncbi.nlm.nih.gov/pubmed/12949182
Microbiology. 2003 Sep;149(Pt 9):2585-96.
Gene expression level influences amino acid usage, but not codon usage, in the tsetse fly endosymbiont Wigglesworthia.

https://www.ncbi.nlm.nih.gov/pubmed/12200484
Mol Biol Evol. 2002 Sep;19(9):1575-84.
A strong effect of AT mutational bias on amino acid usage in Buchnera is mitigated at high-expression genes.
Palacios C1, Wernegreen JJ.

https://www.ncbi.nlm.nih.gov/pubmed/11305938
Genome Biol. 2001;2(4):RESEARCH0010. Epub 2001 Mar 22.
A simple model based on mutation and selection explains trends in codon and amino-acid usage and GC composition within and across genomes.
Knight RD1, Freeland SJ, Landweber LF.

https://www.ncbi.nlm.nih.gov/pubmed/11070046
Mol Biol Evol. 2000 Nov;17(11):1581-8.
Nucleotide bias causes a genomewide bias in the amino acid composition of proteins.
Singer GA1, Hickey DA.

----------

### Growth rate

https://www.ncbi.nlm.nih.gov/pubmed/20308095
Philos Trans R Soc Lond B Biol Sci. 2010 Apr 27;365(1544):1203-12. doi: 10.1098/rstb.2009.0305.
Forces that influence the evolution of codon bias.
Sharp PM1, Emery LR, Zeng K.
- Here we show that the strength of selected codon usage bias is highly correlated with bacterial growth rate, suggesting that selection has favoured translational efficiency. 

https://www.ncbi.nlm.nih.gov/pubmed/15728743
Nucleic Acids Res. 2005 Feb 23;33(4):1141-53. Print 2005.
Variation in the strength of selected codon usage bias among bacteria.
Sharp PM1, Bailes E, Grocock RJ, Peden JF, Sockett RE.
- These results are consistent with the hypothesis that species exposed to selection for rapid growth have more rRNA operons, more tRNA genes and more strongly selected codon usage bias. For example, Clostridium perfringens, the species with the highest value of S, can have a generation time as short as 7 min.

https://www.ncbi.nlm.nih.gov/pubmed/20090831
PLoS Genet. 2010 Jan 15;6(1):e1000808. doi: 10.1371/journal.pgen.1000808.
The systemic imprint of growth and its uses in ecological (meta)genomics.
Vieira-Silva S1, Rocha EP.

https://www.ncbi.nlm.nih.gov/pubmed/15479947
Genome Res. 2004 Nov;14(11):2279-86. Epub 2004 Oct 12.
Codon usage bias from tRNA's point of view: redundancy, specialization, and efficient decoding for translation optimization.
Rocha EP1.
- We show that as minimal generation times get shorter, the genomes contain more tRNA genes, but fewer anticodon species. 
- It also provides new evidence that a selective force for the optimization of the translation machinery is the maximization of growth.

### Metagenome

https://www.ncbi.nlm.nih.gov/pubmed/27115650
Methods Mol Biol. 2016;1415:509-31. doi: 10.1007/978-1-4939-3572-7_26.
Big Data, Evolution, and Metagenomes: Predicting Disease from Gut Microbiota Codon Usage Profiles.
Fabijanić M1, Vlahoviček K2.

https://phylogenomics.blogspot.jp/2013/08/great-use-of-metagenomic-data-community.html
August 09, 2013
Great use of metagenomic data: community wide adaptation signatures

https://www.ncbi.nlm.nih.gov/pubmed/23921637
Nucleic Acids Res. 2013 Oct;41(19):8842-52. doi: 10.1093/nar/gkt673. Epub 2013 Aug 5.
Environmental shaping of codon usage and functional adaptation across microbial communities.
Roller M1, Lucić V, Nagy I, Perica T, Vlahovicek K.

### Database

https://www.ncbi.nlm.nih.gov/pubmed/22536831
BMC Bioinformatics. 2012 Apr 26;13:62. doi: 10.1186/1471-2105-13-62.
CBDB: the codon bias database.
Hilterbrand A1, Saelens J, Putonti C.


### Tools

https://www.ncbi.nlm.nih.gov/pubmed/15980527
Nucleic Acids Res. 2005 Jul 1;33(Web Server issue):W526-31.
JCat: a novel tool to adapt codon usage of a target gene to its potential expression host.
Grote A1, Hiller K, Scheer M, Münch R, Nörtemann B, Hempel DC, Jahn D.

http://codonw.sourceforge.net
Correspondence Analysis of Codon Usage
CodonW
15/Apr/2005

http://pbil.univ-lyon1.fr/datasets/charif04/
Online Synonymous Codon Usage Analyses with the ade4 and seqinR packages
2004

### Temperature

https://www.ncbi.nlm.nih.gov/pubmed/16989961
Gene. 2006 Dec 30;385:128-36. 
Synonymous codon usage and its potential link with optimal growth temperature in prokaryotes.
Lobry JR1, Necşulea A.

https://www.ncbi.nlm.nih.gov/pubmed/12364606
Nucleic Acids Res. 2002 Oct 1;30(19):4272-7.
Synonymous codon usage is subject to selection in thermophilic bacteria.
Lynn DJ1, Singer GA, Hickey DA.

### Review
http://www.ncbi.nlm.nih.gov/pubmed/21102527
Nat Rev Genet. 2011 Jan;12(1):32-42. doi: 10.1038/nrg2899. Epub 2010 Nov 23.
Synonymous but not the same: the causes and consequences of codon bias.
Plotkin JB1, Kudla G.

https://www.ncbi.nlm.nih.gov/pubmed/11719972
Curr Issues Mol Biol. 2001 Oct;3(4):91-7.
Synonymous codon usage in bacteria.
Ermolaeva MD1.

### Olsen

https://www.youtube.com/watch?v=gzYCZ80OygU
The Alien-Looking Codon Usages of Recently Acquired Genes Are Not Alien - Gary Olsen - YouTube
Published on Oct 7, 2015

https://www.ncbi.nlm.nih.gov/pubmed/22128332
Proc Natl Acad Sci U S A. 2011 Dec 13;108(50):20154-9. doi: 10.1073/pnas.1109451108. Epub 2011 Nov 29.
Similarity of genes horizontally acquired by Escherichia coli and Salmonella enterica is evidence of a supraspecies pangenome.
Karberg KA1, Olsen GJ, Davis JJ.

https://www.ncbi.nlm.nih.gov/pubmed/20679093
Mol Biol Evol. 2011 Jan;28(1):211-21. doi: 10.1093/molbev/msq185. Epub 2010 Aug 2.
Characterizing the native codon usages of a genome: an axis projection approach.
Davis JJ1, Olsen GJ.

https://www.ncbi.nlm.nih.gov/pubmed/20018979
Mol Biol Evol. 2010 Apr;27(4):800-10. doi: 10.1093/molbev/msp281. Epub 2009 Dec 17.
Modal codon usage: assessing the typical codon usage of a genome.
Davis JJ1, Olsen GJ.

### Plasmids

https://www.ncbi.nlm.nih.gov/pubmed/19221094
DNA Res. 2009 Apr;16(2):91-104. doi: 10.1093/dnares/dsp001. Epub 2009 Feb 15.
Codon usages of genes on chromosome, and surprisingly, genes in plasmid are primarily affected by strand-specific mutational biases in Lawsonia intracellularis.
Guo FB1, Yuan JB.

https://www.ncbi.nlm.nih.gov/pubmed/18391244
In Silico Biol. 2007;7(4-5):547-58.
Analysis of codon usage patterns and predicted highly expressed genes for six phytopathogenic Xanthomonas genomes shows a high degree of conservation.
Sen G1, Sur S, Bose D, Mondal U, Furnholm T, Bothra A, Tisa L, Sen A.

https://www.ncbi.nlm.nih.gov/pubmed/14597394
Gene. 2003 Nov 27;320:109-16.
The strength of translational selection for codon usage varies in the three replicons of Sinorhizobium meliloti.
Peixoto L1, Zavala A, Romero H, Musto H.

https://www.ncbi.nlm.nih.gov/pubmed/9820537
J Immunol. 1998 Nov 15;161(10):5594-9.
Optimization of codon usage of plasmid DNA vaccine is required for the effective MHC class I-restricted T cell responses against an intracellular bacterium.
Uchijima M1, Yoshida A, Nagata T, Koide Y.


### Virus

https://www.ncbi.nlm.nih.gov/pubmed/27278133
Sci Rep. 2016 Jun 9;6:27546. doi: 10.1038/srep27546.
Codon optimization of the adenoviral fiber negatively impacts structural protein expression and viral fitness.
Villanueva E1, Martí-Solano M2, Fillat C1,3.

https://www.ncbi.nlm.nih.gov/pubmed/22016848
Mob Genet Elements. 2011 May;1(1):75-77.
Codon bias, tRNA pools and horizontal gene transfer.
Tuller T1.

https://www.ncbi.nlm.nih.gov/pubmed/19888206
Mol Syst Biol. 2009;5:311. doi: 10.1038/msb.2009.71. Epub 2009 Oct 13.
Viral adaptation to host: a proteome-based analysis of codon usage and amino acid preferences.
Bahir I1, Fromer M, Prat Y, Linial M.



### Video

https://www.youtube.com/watch?v=P-fjZPf3Dnw
Codon optimization: Why & how to design DNA sequences for optimal soluble protein expression - YouTube
Published on Nov 3, 2014

### Musto

https://www.ncbi.nlm.nih.gov/pubmed/23288542
Microbiology. 2013 Mar;159(Pt 3):555-64.
Evolution of optimal codon choices in the family Enterobacteriaceae.
Iriarte A1, Baraibar JD, Romero H, Castro-Sowinski S, Musto H.

### Sharp

https://www.ncbi.nlm.nih.gov/pubmed/8662004
J Mol Evol. 1996 May;42(5):525-36.
Codon usage and base composition in Rickettsia prowazekii.
Andersson SG1, Sharp PM.

### Kanaya

https://www.ncbi.nlm.nih.gov/pubmed/11591475
Gene. 2001 Oct 3;276(1-2):89-99.
Analysis of codon usage diversity of bacterial genes with a self-organizing map (SOM): characterization of horizontally transferred genes with emphasis on the E. coli O157 genome.
Kanaya S1, Kinouchi M, Abe T, Kudo Y, Yamada Y, Nishi T, Mori H, Ikemura T.

https://www.ncbi.nlm.nih.gov/pubmed/10570992
Gene. 1999 Sep 30;238(1):143-55.
Studies of codon usage and tRNA genes of 18 unicellular organisms and quantification of Bacillus subtilis tRNAs: gene expression level and species-specific diversity of codon usage based on multivariate analysis.
Kanaya S1, Yamada Y, Kudo Y, Ikemura T.

### Suzuki

http://www.ncbi.nlm.nih.gov/pubmed/27196606
PLoS One. 2016 May 19;11(5):e0154306.
Codon Adaptation of Plastid Genes.
Suzuki H, Morton BR.

http://www.ncbi.nlm.nih.gov/pubmed/26341535
FEBS Lett. 2015 Oct 7;589(20 Pt B):3113-8.
Analysis of amino acid and codon usage in Paramecium bursaria.
Dohra H, Fujishima M, Suzuki H.

http://www.ncbi.nlm.nih.gov/pubmed/19480720
BMC Bioinformatics. 2009 Jun 1;10:167.
Measure of synonymous codon usage diversity among genes in bacteria.
Suzuki H1, Saito R, Tomita M.

http://www.ncbi.nlm.nih.gov/pubmed/18940873
DNA Res. 2008 Dec;15(6):357-65.
Comparison of correspondence analysis methods for synonymous codon usage in bacteria.
Suzuki H, Brown CJ, Forney LJ, Top EM.

http://www.ncbi.nlm.nih.gov/pubmed/18350114
EURASIP J Bioinform Syst Biol. 2007:61374.
Variation in the correlation of G + C composition with synonymous codon usage bias among bacteria.
Suzuki H1, Saito R, Tomita M.

http://www.ncbi.nlm.nih.gov/pubmed/16289058
FEBS Lett. 2005 Nov 21;579(28):6499-504.
A problem in multivariate analysis of codon usage data and a possible solution.
Suzuki H1, Saito R, Tomita M.

http://www.ncbi.nlm.nih.gov/pubmed/15194186
Gene. 2004 Jun 23;335:19-23.
The 'weighted sum of relative entropy': a new index for synonymous codon usage bias.
Suzuki H1, Saito R, Tomita M.

### Unclassified

https://www.ncbi.nlm.nih.gov/pubmed/18983258
Annu Rev Genet. 2008;42:287-99. doi: 10.1146/annurev.genet.42.110807.091442.
Selection on codon bias.
Hershberg R1, Petrov DA.

https://www.ncbi.nlm.nih.gov/pubmed/22032172
Genome Biol. 2011 Oct 27;12(10):R109. doi: 10.1186/gb-2011-12-10-r109.
Variation in global codon usage bias among prokaryotic organisms is associated with their lifestyles.
Botzman M1, Margalit H.

https://www.ncbi.nlm.nih.gov/pubmed/21343180
Nucleic Acids Res. 2011 Jun;39(11):4743-55. doi: 10.1093/nar/gkr054. Epub 2011 Feb 22.
Association between translation efficiency and horizontal gene transfer within microbial communities.
Tuller T1, Girshovich Y, Sella Y, Kreimer A, Freilich S, Kupiec M, Gophna U, Ruppin E.

https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.

https://www.ncbi.nlm.nih.gov/pubmed/15537809
Mol Biol Evol. 2005 Mar;22(3):547-61. Epub 2004 Nov 10.
Codon bias signatures, organization of microorganisms in codon space, and lifestyle.
Carbone A1, Képès F, Zinovyev A.

https://www.ncbi.nlm.nih.gov/pubmed/14594704
Bioinformatics. 2003 Nov 1;19(16):2005-15.
Codon adaptation index as a measure of dominating codon bias.
Carbone A1, Zinovyev A, Képès F.

https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.

### Japanese

http://tenure5.vbl.okayama-u.ac.jp/HM_blog/?s=コドン
コドン | 酵母とシステムバイオロジー

2017/1/18
http://yfuruta.sakura.ne.jp/blog/?p=824
コドン最適化

2013年 第11号
https://www.sbj.or.jp/wp-content/uploads/file/sbj/9111/9111_biomedia_2(1).pdf
コドンが決め手!?組換えタンパク質の高発現法 今中 洋行

2010-08-25
http://d.hatena.ne.jp/morimori-08/20100825/1282745255
コドンバイアスはタンパク質のフォールディングのエラーを少なくする

https://www.nig.ac.jp/museum/evolution/04.html
遺伝暗号(コドン）使用の種による多様性

https://biowiki.edu-wiki.org/コドン表
コドン表 [BioWiki]

https://integbio.jp/dbcatalog/record/nbdc00033
Codon Usage Database - Integbio データベースカタログ

http://www.kazusa.or.jp/codon/
Codon Usage Database

http://www.genscript.jp/codon_opt.html
OptimumGene™コドン最適化技術 - GenScript

----------

[コドン - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20130122)

[論文 - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20090605)

http://www.kri.sfc.keio.ac.jp/report/mori/2005/b-13.html
研究課題名        バイオインフォマティックスによる遺伝暗号解析

