----------

Haruo Suzuki (haruo[at]g-language[dot]org)  
Last Update: 2017-07-10

----------

# Codon usage

----------

## References

### Video

https://www.youtube.com/watch?v=gzYCZ80OygU
The Alien-Looking Codon Usages of Recently Acquired Genes Are Not Alien - Gary Olsen - YouTube
Published on Oct 7, 2015

https://www.youtube.com/watch?v=P-fjZPf3Dnw
Codon optimization: Why & how to design DNA sequences for optimal soluble protein expression - YouTube
Published on Nov 3, 2014

### Metagenome

https://www.ncbi.nlm.nih.gov/pubmed/27115650
Methods Mol Biol. 2016;1415:509-31. doi: 10.1007/978-1-4939-3572-7_26.
Big Data, Evolution, and Metagenomes: Predicting Disease from Gut Microbiota Codon Usage Profiles.
Fabijanić M1, Vlahoviček K2.

https://phylogenomics.blogspot.jp/2013/08/great-use-of-metagenomic-data-community.html
August 09, 2013
Great use of metagenomic data: community wide adaptation signatures

https://www.ncbi.nlm.nih.gov/pubmed/23921637
Nucleic Acids Res. 2013 Oct;41(19):8842-52. doi: 10.1093/nar/gkt673. Epub 2013 Aug 5.
Environmental shaping of codon usage and functional adaptation across microbial communities.
Roller M1, Lucić V, Nagy I, Perica T, Vlahovicek K.

### Growth rate

https://www.ncbi.nlm.nih.gov/pubmed/20308095
Philos Trans R Soc Lond B Biol Sci. 2010 Apr 27;365(1544):1203-12. doi: 10.1098/rstb.2009.0305.
Forces that influence the evolution of codon bias.
Sharp PM1, Emery LR, Zeng K.
- Here we show that the strength of selected codon usage bias is highly correlated with bacterial growth rate, suggesting that selection has favoured translational efficiency. 

https://www.ncbi.nlm.nih.gov/pubmed/15728743
Nucleic Acids Res. 2005 Feb 23;33(4):1141-53. Print 2005.
Variation in the strength of selected codon usage bias among bacteria.
Sharp PM1, Bailes E, Grocock RJ, Peden JF, Sockett RE.
- These results are consistent with the hypothesis that species exposed to selection for rapid growth have more rRNA operons, more tRNA genes and more strongly selected codon usage bias. For example, Clostridium perfringens, the species with the highest value of S, can have a generation time as short as 7 min.

https://www.ncbi.nlm.nih.gov/pubmed/20090831
PLoS Genet. 2010 Jan 15;6(1):e1000808. doi: 10.1371/journal.pgen.1000808.
The systemic imprint of growth and its uses in ecological (meta)genomics.
Vieira-Silva S1, Rocha EP.

https://www.ncbi.nlm.nih.gov/pubmed/15479947
Genome Res. 2004 Nov;14(11):2279-86. Epub 2004 Oct 12.
Codon usage bias from tRNA's point of view: redundancy, specialization, and efficient decoding for translation optimization.
Rocha EP1.
- We show that as minimal generation times get shorter, the genomes contain more tRNA genes, but fewer anticodon species. 
- It also provides new evidence that a selective force for the optimization of the translation machinery is the maximization of growth.

### Database

https://www.ncbi.nlm.nih.gov/pubmed/22536831
BMC Bioinformatics. 2012 Apr 26;13:62. doi: 10.1186/1471-2105-13-62.
CBDB: the codon bias database.
Hilterbrand A1, Saelens J, Putonti C.


### Tools

http://codonw.sourceforge.net
Correspondence Analysis of Codon Usage
CodonW
15/Apr/2005

http://pbil.univ-lyon1.fr/datasets/charif04/
Online Synonymous Codon Usage Analyses with the ade4 and seqinR packages
2004

### Temperature

https://www.ncbi.nlm.nih.gov/pubmed/16989961
Gene. 2006 Dec 30;385:128-36. 
Synonymous codon usage and its potential link with optimal growth temperature in prokaryotes.
Lobry JR1, Necşulea A.

https://www.ncbi.nlm.nih.gov/pubmed/12364606
Nucleic Acids Res. 2002 Oct 1;30(19):4272-7.
Synonymous codon usage is subject to selection in thermophilic bacteria.
Lynn DJ1, Singer GA, Hickey DA.

### Amino acid usage

https://www.ncbi.nlm.nih.gov/pubmed/8065933
Nucleic Acids Res. 1994 Aug 11;22(15):3174-80.
Hydrophobicity, expressivity and aromaticity are the major trends of amino-acid usage in 999 Escherichia coli chromosome-encoded genes.
Lobry JR1, Gautier C.

### Review
http://www.ncbi.nlm.nih.gov/pubmed/21102527
Nat Rev Genet. 2011 Jan;12(1):32-42. doi: 10.1038/nrg2899. Epub 2010 Nov 23.
Synonymous but not the same: the causes and consequences of codon bias.
Plotkin JB1, Kudla G.

https://www.ncbi.nlm.nih.gov/pubmed/11719972
Curr Issues Mol Biol. 2001 Oct;3(4):91-7.
Synonymous codon usage in bacteria.
Ermolaeva MD1.

### Unclassified

https://www.ncbi.nlm.nih.gov/pubmed/22032172
Genome Biol. 2011 Oct 27;12(10):R109. doi: 10.1186/gb-2011-12-10-r109.
Variation in global codon usage bias among prokaryotic organisms is associated with their lifestyles.
Botzman M1, Margalit H.

https://www.ncbi.nlm.nih.gov/pubmed/21343180
Nucleic Acids Res. 2011 Jun;39(11):4743-55. doi: 10.1093/nar/gkr054. Epub 2011 Feb 22.
Association between translation efficiency and horizontal gene transfer within microbial communities.
Tuller T1, Girshovich Y, Sella Y, Kreimer A, Freilich S, Kupiec M, Gophna U, Ruppin E.

https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.

https://www.ncbi.nlm.nih.gov/pubmed/15537809
Mol Biol Evol. 2005 Mar;22(3):547-61. Epub 2004 Nov 10.
Codon bias signatures, organization of microorganisms in codon space, and lifestyle.
Carbone A1, Képès F, Zinovyev A.

https://www.ncbi.nlm.nih.gov/pubmed/14594704
Bioinformatics. 2003 Nov 1;19(16):2005-15.
Codon adaptation index as a measure of dominating codon bias.
Carbone A1, Zinovyev A, Képès F.


https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.




### Plasmids
- [Davis JJ, Olsen GJ. Mol Biol Evol. 2010 Apr "Modal codon usage: assessing the typical codon usage of a genome."](https://www.ncbi.nlm.nih.gov/pubmed/20018979)
- [Guo FB, Yuan JB. DNA Res. 2009 Apr "Codon usages of genes on chromosome, and surprisingly, genes in plasmid are primarily affected by strand-specific mutational biases in Lawsonia intracellularis."](https://www.ncbi.nlm.nih.gov/pubmed/19221094)
- [Peixoto L et al. Gene. 2003 Nov 27 "The strength of translational selection for codon usage varies in the three replicons of Sinorhizobium meliloti."](https://www.ncbi.nlm.nih.gov/pubmed/14597394)

----------

[コドン - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20130122)

[論文 - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20090605)

http://www.kri.sfc.keio.ac.jp/report/mori/2005/b-13.html
研究課題名        バイオインフォマティックスによる遺伝暗号解析

