Haruo Suzuki (haruo[at]g-language[dot]org)  
Last Update: 2018-10

----------

# Codon usage

## Table of Contents
- [featuring](#featuring)
- [lifestyle](#lifestyle)
- [updates](#updates)
  - [2018](#2018)
  - [2017](#2017)
- [database](#database)
- [tools](#tools)
- [mge](#mge)
  - [plasmids](#plasmids)
  - [virus](#virus)
- [review](#review)
- [metagenome](#metagenome)
- [growth rate](#growth-rate)
- [temperature](#temperature)
- [replication](#replication)
- [people](#people)
  - [Kanaya](#kanaya)
  - [Musto](#musto)
  - [Olsen](#olsen)
  - [Sharp](#sharp)
  - [Suzuki](#suzuki)
- [unclassified](#unclassified)

----------
## featuring

http://www.oxfordscholarship.com/view/10.1093/acprof:osobl/9780199601165.001.0001/acprof-9780199601165
Codon Evolution: Mechanisms and Models - Oxford Scholarship
http://wp.biota.utoronto.ca/chang/files/2016/02/Chang-et-al-Codon-Evolution-Chapter11.pdf

https://www.ncbi.nlm.nih.gov/pubmed/20369015
PLoS Comput Biol. 2010 Apr 1;6(4):e1000732. doi: 10.1371/journal.pcbi.1000732.
Why genes evolve faster on secondary chromosomes in bacteria.
Cooper VS1, Vohr SH, Wrocklage SC, Hatcher PJ.
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2848543/

----------

### lifestyle

https://www.ncbi.nlm.nih.gov/pubmed/22032172
Genome Biol. 2011 Oct 27;12(10):R109. doi: 10.1186/gb-2011-12-10-r109.
Variation in global codon usage bias among prokaryotic organisms is associated with their lifestyles.
Botzman M1, Margalit H.

https://www.ncbi.nlm.nih.gov/pubmed/23024607
Curr Genomics. 2012 Apr;13(2):153-62.
Microbial lifestyle and genome signatures.
Dutta C1, Paul S.
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3308326/
c) Codon Usage
Distinct niche-specific trends in synonymous codon usage have also been observed in microbes thriving at high salinity [24]. An analysis of synonymous codon usage patterns in bacterial and fungal genomes by Willenbrok et al. [49] demonstrated that differences in codon preferences of translational codon adaptation and dominant codon adaptation provide an environmental signature that can segregate bacteria according to their lifestyle, for instance soil bacteria and soil symbionts, spore formers, enteric bacteria, aquatic bacteria, and small intercellular and extracellular pathogens.

https://www.ncbi.nlm.nih.gov/pubmed/17156429
Genome Biol. 2006;7(12):R114.
An environmental signature for 323 microbial genomes based on codon adaptation indices.
Willenbrock H1, Friis C, Juncker AS, Ussery DW.
group bacteria according to their lifestyle, 
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1794427/
Here, we calculate and compare a translational CAI (tCAI) based on that proposed by Sharp and Li [1] with a purely mathematical dominant CAI (dCAI) [8] for 318 bacterial and 5 fungal genomes for which full sequences are deposited in Genbank and available from the Genome Atlas Database (version 19.1) [11].
It was previously postulated that fast-growing bacteria share codon usage preferences because they have more abundant and similar tRNAs [12]. 

https://www.ncbi.nlm.nih.gov/pubmed/15537809
Mol Biol Evol. 2005 Mar;22(3):547-61. Epub 2004 Nov 10.
Codon bias signatures, organization of microorganisms in codon space, and lifestyle.
Carbone A1, Képès F, Zinovyev A.

https://www.ncbi.nlm.nih.gov/pubmed/14594704
Bioinformatics. 2003 Nov 1;19(16):2005-15.
Codon adaptation index as a measure of dominating codon bias.
Carbone A1, Zinovyev A, Képès F.

----------

## updates

### 2018

https://www.ncbi.nlm.nih.gov/pubmed/30247489
Nat Biotechnol. 2018 Sep 24. doi: 10.1038/nbt.4238. [Epub ahead of print]
Evaluation of 244,000 synthetic sequences reveals design principles to optimize translation in Escherichia coli.
Cambray G1,2, Guimaraes JC1,3, Arkin AP3,4.

Codon composition has a sizable impact on translatability, but only in comparatively rare elongation-limited transcripts.

https://twitter.com/kfuku0502/status/1003768475871875072
Kenji Fukushima on Twitter: "高発現遺伝子だけでなく低発現遺伝子でもコドン使用頻度は選択を受けているらしい。大腸菌でそういうサイトの同義置換を入れて適応度が下がるケースを発見している。 https://t.co/nFGnUi3m2A"

https://www.ncbi.nlm.nih.gov/pubmed/29688501
Genome Biol Evol. 2018 Apr 1;10(5):1237-1246. doi: 10.1093/gbe/evy084.
The Codon Usage of Lowly Expressed Genes Is Subject to Natural Selection.
Yannai A1, Katz S1, Hershberg R1.


高度発現遺伝子群のコドンバイアスが、他の遺伝子群の翻訳に影響する。
https://www.ncbi.nlm.nih.gov/pubmed/29735666
Proc Natl Acad Sci U S A. 2018 May 22;115(21):E4940-E4949. doi: 10.1073/pnas.1719375115. Epub 2018 May 7.
Codon usage of highly expressed genes affects proteome-wide translation efficiency.
Frumkin I1, Lajoie MJ2, Gregg CJ2, Hornung G3, Church GM4, Pilpel Y5.

### 2017

----------
## database

https://www.ncbi.nlm.nih.gov/pubmed/22536831
BMC Bioinformatics. 2012 Apr 26;13:62. doi: 10.1186/1471-2105-13-62.
CBDB: the codon bias database.
Hilterbrand A1, Saelens J, Putonti C.

## tools

22 Aug 2018
https://arxiv.org/abs/1808.07259
[1808.07259] SSCU: an R/Bioconductor package for analyzing selective profile in synonymous codon usage

異種遺伝子発現のためのコドン改変
https://galaxyproject.org/public-galaxy-servers/codon-harmonizer/
Codon Harmonizer
https://www.ncbi.nlm.nih.gov/pubmed/28902855
PLoS One. 2017 Sep 13;12(9):e0184355. doi: 10.1371/journal.pone.0184355. eCollection 2017.
Improving heterologous membrane protein production in Escherichia coli by combining transcriptional tuning and codon usage algorithms.

https://www.ncbi.nlm.nih.gov/pubmed/17439967
Nucleic Acids Res. 2007 Jul;35(Web Server issue):W126-31. Epub 2007 Apr 16.
OPTIMIZER: a web server for optimizing the codon usage of DNA sequences.
Puigbò P1, Guzmán E, Romeu A, Garcia-Vallvé S.

https://www.ncbi.nlm.nih.gov/pubmed/15980527
Nucleic Acids Res. 2005 Jul 1;33(Web Server issue):W526-31.
JCat: a novel tool to adapt codon usage of a target gene to its potential expression host.
Grote A1, Hiller K, Scheer M, Münch R, Nörtemann B, Hempel DC, Jahn D.

http://codonw.sourceforge.net
Correspondence Analysis of Codon Usage
CodonW
15/Apr/2005

http://pbil.univ-lyon1.fr/datasets/charif04/
Online Synonymous Codon Usage Analyses with the ade4 and seqinR packages
2004

http://bioinfo.ie.niigata-u.ac.jp/?Codon%20Usage%20Generator
Codon Usage Generator (CUG) †
本プログラムは，コドン組成を計算するためのソフトウェアです．

http://bioinfo.ie.niigata-u.ac.jp/?遺伝子発現量予測

----------
## mge
Mobile genetic elements (MGEs)
https://ja.wikipedia.org/wiki/可動遺伝因子

### plasmids

https://www.ncbi.nlm.nih.gov/pubmed/28206693
Mol Ecol. 2017 Apr;26(7):1832-1847. doi: 10.1111/mec.14056. Epub 2017 Mar 13.
Plasmid and clonal interference during post horizontal gene transfer evolution.
Bedhomme S1,2, Perez Pantoja D3, Bravo IG2.
KEYWORDS:
antibiotic resistance; clonal interference; codon usage preferences; multilevel selection; plasmid
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5392415/
Plasmid carriage costs are not the only costs to plasmid-mediated HGT. Another potentially important one is the mismatch in codon usage preferences between the transferred gene and the receiving genome (Baltrus 2013). Each species is indeed characterized by specific frequencies of use of the different codons within a synonymous codon family. An important evolutionary force shaping codon usage preferences is coevolution with the translation machinery: codon usage frequencies are strongly related with the copy number of corresponding tRNA genes, especially for highly expressed genes and for rapidly growing organisms (Rocha 2004). Genes transferred horizontally from a different organism can present a mismatch in codon usage preferences with the receiving genome, and such mismatch is known to affect translation accuracy and speed (Komar et al. 1999, Burgess-Brown et al. 2008). 

and comparative approaches demonstrate that HGT is more likely to be successful between genomes with similar codon usage preferences (Tuller et al. 2011, Medrano-Soto 2004).

The cost of plasmid carriage was quantified and we showed that mismatches in codon usage preferences generate differences in chloramphenicol resistance. 

https://www.ncbi.nlm.nih.gov/pubmed/19221094
DNA Res. 2009 Apr;16(2):91-104. doi: 10.1093/dnares/dsp001. Epub 2009 Feb 15.
Codon usages of genes on chromosome, and surprisingly, genes in plasmid are primarily affected by strand-specific mutational biases in Lawsonia intracellularis.
Guo FB1, Yuan JB.

https://www.ncbi.nlm.nih.gov/pubmed/18391244
In Silico Biol. 2007;7(4-5):547-58.
Analysis of codon usage patterns and predicted highly expressed genes for six phytopathogenic Xanthomonas genomes shows a high degree of conservation.
Sen G1, Sur S, Bose D, Mondal U, Furnholm T, Bothra A, Tisa L, Sen A.

https://www.ncbi.nlm.nih.gov/pubmed/14597394
Gene. 2003 Nov 27;320:109-16.
The strength of translational selection for codon usage varies in the three replicons of Sinorhizobium meliloti.
Peixoto L1, Zavala A, Romero H, Musto H.

https://www.ncbi.nlm.nih.gov/pubmed/9820537
J Immunol. 1998 Nov 15;161(10):5594-9.
Optimization of codon usage of plasmid DNA vaccine is required for the effective MHC class I-restricted T cell responses against an intracellular bacterium.
Uchijima M1, Yoshida A, Nagata T, Koide Y.

### virus

https://www.ncbi.nlm.nih.gov/pubmed/27278133
Sci Rep. 2016 Jun 9;6:27546. doi: 10.1038/srep27546.
Codon optimization of the adenoviral fiber negatively impacts structural protein expression and viral fitness.
Villanueva E1, Martí-Solano M2, Fillat C1,3.

https://www.ncbi.nlm.nih.gov/pubmed/22016848
Mob Genet Elements. 2011 May;1(1):75-77.
Codon bias, tRNA pools and horizontal gene transfer.
Tuller T1.

https://www.ncbi.nlm.nih.gov/pubmed/19888206
Mol Syst Biol. 2009;5:311. doi: 10.1038/msb.2009.71. Epub 2009 Oct 13.
Viral adaptation to host: a proteome-based analysis of codon usage and amino acid preferences.
Bahir I1, Fromer M, Prat Y, Linial M.

https://www.ncbi.nlm.nih.gov/pubmed/18463708
PLoS Comput Biol. 2008 Feb 29;4(2):e1000001. doi: 10.1371/journal.pcbi.1000001.
Genome landscapes and bacteriophage codon usage.
Lucks JB1, Nelson DR, Kudla GR, Plotkin JB.


----------
### Unclassified

https://www.ncbi.nlm.nih.gov/pubmed/21343180
Nucleic Acids Res. 2011 Jun;39(11):4743-55. doi: 10.1093/nar/gkr054. Epub 2011 Feb 22.
Association between translation efficiency and horizontal gene transfer within microbial communities.
Tuller T1, Girshovich Y, Sella Y, Kreimer A, Freilich S, Kupiec M, Gophna U, Ruppin E.


----------
### review

https://www.ncbi.nlm.nih.gov/pubmed/29018283
Nat Rev Mol Cell Biol. 2017 Oct 11.
Codon optimality, bias and usage in translation and mRNA decay.
Hanson G1, Coller J1.

https://www.ncbi.nlm.nih.gov/pubmed/26186290
Mol Cell. 2015 Jul 16;59(2):149-61. doi: 10.1016/j.molcel.2015.05.035.
Codon Bias as a Means to Fine-Tune Gene Expression.
Quax TE1, Claassens NJ2, Söll D3, van der Oost J4.

https://www.ncbi.nlm.nih.gov/pubmed/22889422
Biol Rev Camb Philos Soc. 2013 Feb;88(1):49-61. doi: 10.1111/j.1469-185X.2012.00242.x. Epub 2012 Aug 14.
Codon usage bias: causative factors, quantification methods and genome-wide patterns: with emphasis on insect genomes.
Behura SK1, Severson DW.

https://www.ncbi.nlm.nih.gov/pubmed/22921354
Trends Genet. 2012 Nov;28(11):574-81. doi: 10.1016/j.tig.2012.07.006. Epub 2012 Aug 23.
Speeding with control: codon usage, tRNAs, and ribosomes.
Novoa EM1, Ribas de Pouplana L.

http://www.ncbi.nlm.nih.gov/pubmed/21102527
Nat Rev Genet. 2011 Jan;12(1):32-42. doi: 10.1038/nrg2899. Epub 2010 Nov 23.
Synonymous but not the same: the causes and consequences of codon bias.
Plotkin JB1, Kudla G.
http://mathbio.sas.upenn.edu/Papers/Plotkin_Kudla_NatRevGenetics_2011.pdf

https://www.ncbi.nlm.nih.gov/pubmed/18983258
Annu Rev Genet. 2008;42:287-99. doi: 10.1146/annurev.genet.42.110807.091442.
Selection on codon bias.
Hershberg R1, Petrov DA.

https://www.ncbi.nlm.nih.gov/pubmed/11719972
Curr Issues Mol Biol. 2001 Oct;3(4):91-7.
Synonymous codon usage in bacteria.
Ermolaeva MD1.

----------

### metagenome

https://www.ncbi.nlm.nih.gov/pubmed/29947757
Bioinformatics. 2018 Jun 27. doi: 10.1093/bioinformatics/bty519. [Epub ahead of print]
BMC3C: Binning Metagenomic Contigs using Codon usage, sequence Composition and read Coverage.
Yu G1, Jiang Y1, Wang J1, Zhang H2, Luo H2.

https://www.ncbi.nlm.nih.gov/pubmed/29742107
PLoS One. 2018 May 9;13(5):e0195869. doi: 10.1371/journal.pone.0195869. eCollection 2018.
Codon usage bias reveals genomic adaptations to environmental conditions in an acidophilic consortium.
Hart A1, Cortés MP2,3, Latorre M2,3,4,5, Martinez S6.

Anamaria Elek (2018)
https://repozitorij.unizg.hr/en/islandora/object/pmf%3A4815
coRdon: an R package for codon usage analysis and prediction of gene expressivity | University of Zagreb Repository

https://www.ncbi.nlm.nih.gov/pubmed/27115650
Methods Mol Biol. 2016;1415:509-31.
Big Data, Evolution, and Metagenomes: Predicting Disease from Gut Microbiota Codon Usage Profiles.
Fabijanić M1, Vlahoviček K2.

https://phylogenomics.blogspot.jp/2013/08/great-use-of-metagenomic-data-community.html
August 09, 2013
Great use of metagenomic data: community wide adaptation signatures

https://www.ncbi.nlm.nih.gov/pubmed/23921637
Nucleic Acids Res. 2013 Oct;41(19):8842-52.
Environmental shaping of codon usage and functional adaptation across microbial communities.
Roller M1, Lucić V, Nagy I, Perica T, Vlahovicek K.

https://www.ncbi.nlm.nih.gov/pubmed/20090831
PLoS Genet. 2010 Jan 15;6(1):e1000808. doi: 10.1371/journal.pgen.1000808.
The systemic imprint of growth and its uses in ecological (meta)genomics.
Vieira-Silva S1, Rocha EP.

https://www.ncbi.nlm.nih.gov/pubmed/16200051
EMBO Rep. 2005 Dec;6(12):1208-13.
Environments shape the nucleotide composition of genomes.
Foerstner KU1, von Mering C, Hooper SD, Bork P.

### growth rate

https://www.ncbi.nlm.nih.gov/pubmed/20308095
Philos Trans R Soc Lond B Biol Sci. 2010 Apr 27;365(1544):1203-12. doi: 10.1098/rstb.2009.0305.
Forces that influence the evolution of codon bias.
Sharp PM1, Emery LR, Zeng K.
- Here we show that the strength of selected codon usage bias is highly correlated with bacterial growth rate, suggesting that selection has favoured translational efficiency. 

https://www.ncbi.nlm.nih.gov/pubmed/15479947
Genome Res. 2004 Nov;14(11):2279-86. Epub 2004 Oct 12.
Codon usage bias from tRNA's point of view: redundancy, specialization, and efficient decoding for translation optimization.
Rocha EP1.
- We show that as minimal generation times get shorter, the genomes contain more tRNA genes, but fewer anticodon species. 
- It also provides new evidence that a selective force for the optimization of the translation machinery is the maximization of growth.

### tRNA


https://www.ncbi.nlm.nih.gov/pubmed/15448185
Nucleic Acids Res. 2004 Sep 24;32(17):5036-44. Print 2004.
Solving the riddle of codon usage preferences: a test for translational selection.
dos Reis M1, Savva R, Wernisch L.


### temperature

https://www.ncbi.nlm.nih.gov/pubmed/16989961
Gene. 2006 Dec 30;385:128-36. 
Synonymous codon usage and its potential link with optimal growth temperature in prokaryotes.
Lobry JR1, Necşulea A.

https://www.ncbi.nlm.nih.gov/pubmed/12364606
Nucleic Acids Res. 2002 Oct 1;30(19):4272-7.
Synonymous codon usage is subject to selection in thermophilic bacteria.
Lynn DJ1, Singer GA, Hickey DA.

----------
### replication

https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.

https://www.ncbi.nlm.nih.gov/pubmed/9724767
Proc Natl Acad Sci U S A. 1998 Sep 1;95(18):10698-703.
Replicational and transcriptional selection on codon usage in Borrelia burgdorferi.
McInerney JO1.


### Karlin
https://www.ncbi.nlm.nih.gov/pubmed/9928479
Annu Rev Genet. 1998;32:185-225.
Comparative DNA analysis across diverse genomes.
Karlin S1, Campbell AM, Mrázek J.
- CODON BIASES IN BACTERIAL GENOMES
- Anomalies of Ribosomal Proteins

https://www.ncbi.nlm.nih.gov/pubmed/15883368
Proc Natl Acad Sci U S A. 2005 May 17;102(20):7303-8. Epub 2005 May 9.
Predicted highly expressed genes in archaeal genomes.
Karlin S1, Mrázek J, Ma J, Brocchieri L.

[Karlin S (2001) Trends Microbiol. "Detecting anomalous gene clusters and pathogenicity islands in diverse bacterial genomes."](https://www.ncbi.nlm.nih.gov/pubmed/11435108) | [pdf](https://eclass.uoa.gr/modules/document/file.php/D473/Βιβλιογραφία/DNA%20Composition/Karlin_2001.pdf)

http://www.cmbl.uga.edu/software/ASeqH.htm
PHX/PA user guide
Analysis of sequence heterogeneity by sliding window plots

http://www.cmbl.uga.edu/software/PHX-PA-guide.htm
CMBL- PHX/PA user guide

----------

### Amino acid usage

https://www.ncbi.nlm.nih.gov/pubmed/11904428
Proc Natl Acad Sci U S A. 2002 Mar 19;99(6):3695-700.
Metabolic efficiency and amino acid composition in the proteomes of Escherichia coli and Bacillus subtilis.
Akashi H1, Gojobori T.

https://www.ncbi.nlm.nih.gov/pubmed/9461405
Gene. 1997 Dec 31;205(1-2):309-16.
Influence of genomic G+C content on average amino-acid composition of proteins from 59 bacterial species.
Lobry JR1.

https://www.ncbi.nlm.nih.gov/pubmed/8065933
Nucleic Acids Res. 1994 Aug 11;22(15):3174-80.
Hydrophobicity, expressivity and aromaticity are the major trends of amino-acid usage in 999 Escherichia coli chromosome-encoded genes.
Lobry JR1, Gautier C.

#### "amino acid usage" mutational bias

https://www.ncbi.nlm.nih.gov/pubmed/16936139
Mol Biol Evol. 2006 Dec;23(12):2303-15. Epub 2006 Aug 25.
The evolution of biased codon and amino acid usage in nematode genomes.
Cutter AD1, Wasmuth JD, Blaxter ML.

https://www.ncbi.nlm.nih.gov/pubmed/14672975
Genome Res. 2004 Jan;14(1):44-53. Epub 2003 Dec 12.
Mutational and selective pressures on codon and amino acid usage in Buchnera, endosymbiotic bacteria of aphids.
Rispe C1, Delmotte F, van Ham RC, Moya A.

https://www.ncbi.nlm.nih.gov/pubmed/12949182
Microbiology. 2003 Sep;149(Pt 9):2585-96.
Gene expression level influences amino acid usage, but not codon usage, in the tsetse fly endosymbiont Wigglesworthia.

https://www.ncbi.nlm.nih.gov/pubmed/12200484
Mol Biol Evol. 2002 Sep;19(9):1575-84.
A strong effect of AT mutational bias on amino acid usage in Buchnera is mitigated at high-expression genes.
Palacios C1, Wernegreen JJ.

https://www.ncbi.nlm.nih.gov/pubmed/11305938
Genome Biol. 2001;2(4):RESEARCH0010.
A simple model based on mutation and selection explains trends in codon and amino-acid usage and GC composition within and across genomes.
Knight RD1, Freeland SJ, Landweber LF.

https://www.ncbi.nlm.nih.gov/pubmed/11070046
Mol Biol Evol. 2000 Nov;17(11):1581-8.
Nucleotide bias causes a genomewide bias in the amino acid composition of proteins.
Singer GA1, Hickey DA.


----------
## people

### Ikemura
### Kanaya

https://www.ncbi.nlm.nih.gov/pubmed/3916708
Mol Biol Evol. 1985 Jan;2(1):13-34.
Codon usage and tRNA content in unicellular and multicellular organisms.
Ikemura T1.

https://www.ncbi.nlm.nih.gov/pubmed/11675589
J Mol Evol. 2001 Oct-Nov;53(4-5):290-8.
Codon usage and tRNA genes in eukaryotes: correlation of codon usage diversity with translation efficiency and with CG-dinucleotide usage as assessed by multivariate analysis.
Kanaya S1, Yamada Y, Kinouchi M, Kudo Y, Ikemura T.

https://www.ncbi.nlm.nih.gov/pubmed/11591475
Gene. 2001 Oct 3;276(1-2):89-99.
Analysis of codon usage diversity of bacterial genes with a self-organizing map (SOM): characterization of horizontally transferred genes with emphasis on the E. coli O157 genome.
Kanaya S1, Kinouchi M, Abe T, Kudo Y, Yamada Y, Nishi T, Mori H, Ikemura T.

https://www.ncbi.nlm.nih.gov/pubmed/10570992
Gene. 1999 Sep 30;238(1):143-55.
Studies of codon usage and tRNA genes of 18 unicellular organisms and quantification of Bacillus subtilis tRNAs: gene expression level and species-specific diversity of codon usage based on multivariate analysis.
Kanaya S1, Yamada Y, Kudo Y, Ikemura T.

### Musto

https://www.ncbi.nlm.nih.gov/pubmed/23288542
Microbiology. 2013 Mar;159(Pt 3):555-64.
Evolution of optimal codon choices in the family Enterobacteriaceae.
Iriarte A1, Baraibar JD, Romero H, Castro-Sowinski S, Musto H.

https://www.ncbi.nlm.nih.gov/pubmed/10773076
Nucleic Acids Res. 2000 May 15;28(10):2084-90.
Codon usage in Chlamydia trachomatis is the result of strand-specific mutational biases and a complex pattern of selective forces.
Romero H1, Zavala A, Musto H.

### Olsen

http://www.life.illinois.edu/gary/programs/codon_usage.html
Codon Usage Programs

https://www.youtube.com/watch?v=gzYCZ80OygU
The Alien-Looking Codon Usages of Recently Acquired Genes Are Not Alien - Gary Olsen - YouTube
Published on Oct 7, 2015

https://www.ncbi.nlm.nih.gov/pubmed/22128332
Proc Natl Acad Sci U S A. 2011 Dec 13;108(50):20154-9. doi: 10.1073/pnas.1109451108. Epub 2011 Nov 29.
Similarity of genes horizontally acquired by Escherichia coli and Salmonella enterica is evidence of a supraspecies pangenome.
Karberg KA1, Olsen GJ, Davis JJ.

https://www.ncbi.nlm.nih.gov/pubmed/20679093
Mol Biol Evol. 2011 Jan;28(1):211-21. doi: 10.1093/molbev/msq185. Epub 2010 Aug 2.
Characterizing the native codon usages of a genome: an axis projection approach.
Davis JJ1, Olsen GJ.

https://www.ncbi.nlm.nih.gov/pubmed/20018979
Mol Biol Evol. 2010 Apr;27(4):800-10. doi: 10.1093/molbev/msp281. Epub 2009 Dec 17.
Modal codon usage: assessing the typical codon usage of a genome.
Davis JJ1, Olsen GJ.

### Sharp

2011
https://www.era.lib.ed.ac.uk/handle/1842/4893
Codon usage bias in Archaea
Emery, Laura R.

https://www.ncbi.nlm.nih.gov/pubmed/17038449
Mol Biol Evol. 2007 Jan;24(1):10-2. Epub 2006 Oct 12.
Predicting gene expression level from codon usage bias.
Henry I, Sharp PM.

https://www.ncbi.nlm.nih.gov/pubmed/15728743
Nucleic Acids Res. 2005 Feb 23;33(4):1141-53. Print 2005.
Variation in the strength of selected codon usage bias among bacteria.
Sharp PM1, Bailes E, Grocock RJ, Peden JF, Sockett RE.
These results are consistent with the hypothesis that species exposed to selection for rapid growth have more rRNA operons, more tRNA genes and more strongly selected codon usage bias. For example, Clostridium perfringens, the species with the highest value of S, can have a generation time as short as 7 min.
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC549432/

https://www.ncbi.nlm.nih.gov/pubmed/8662004
J Mol Evol. 1996 May;42(5):525-36.
Codon usage and base composition in Rickettsia prowazekii.
Andersson SG1, Sharp PM.

https://www.ncbi.nlm.nih.gov/pubmed/3118331
Nucleic Acids Res. 1987 Oct 12;15(19):8023-40.
Synonymous codon usage in Bacillus subtilis reflects both translational selection and mutational biases.
Shields DC1, Sharp PM.

### Suzuki

http://www.ncbi.nlm.nih.gov/pubmed/27196606
PLoS One. 2016 May 19;11(5):e0154306.
Codon Adaptation of Plastid Genes.
Suzuki H, Morton BR.

http://www.ncbi.nlm.nih.gov/pubmed/26341535
FEBS Lett. 2015 Oct 7;589(20 Pt B):3113-8.
Analysis of amino acid and codon usage in Paramecium bursaria.
Dohra H, Fujishima M, Suzuki H.
ミドリゾウリムシのトランスクリプトーム（RNA-seq）データを用いて、クロレラ共生下でdown-regulatedしている遺伝子群では発現量とアミノ酸・同義コドン使用との間の相関が高い（P. bursaria-Chlorella共生に関わる自然選択？）ことを報告

http://www.ncbi.nlm.nih.gov/pubmed/19480720
BMC Bioinformatics. 2009 Jun 1;10:167.
Measure of synonymous codon usage diversity among genes in bacteria.
Suzuki H1, Saito R, Tomita M.
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2697163/

http://www.ncbi.nlm.nih.gov/pubmed/18940873
DNA Res. 2008 Dec;15(6):357-65.
Comparison of correspondence analysis methods for synonymous codon usage in bacteria.
Suzuki H, Brown CJ, Forney LJ, Top EM.

http://www.ncbi.nlm.nih.gov/pubmed/18350114
EURASIP J Bioinform Syst Biol. 2007:61374.
Variation in the correlation of G + C composition with synonymous codon usage bias among bacteria.
Suzuki H1, Saito R, Tomita M.

http://www.ncbi.nlm.nih.gov/pubmed/16289058
FEBS Lett. 2005 Nov 21;579(28):6499-504.
A problem in multivariate analysis of codon usage data and a possible solution.
Suzuki H1, Saito R, Tomita M.

http://www.ncbi.nlm.nih.gov/pubmed/15194186
Gene. 2004 Jun 23;335:19-23.
The 'weighted sum of relative entropy': a new index for synonymous codon usage bias.
Suzuki H1, Saito R, Tomita M.

----------

### Video

https://www.youtube.com/watch?v=P-fjZPf3Dnw
Codon optimization: Why & how to design DNA sequences for optimal soluble protein expression - YouTube
Published on Nov 3, 2014


----------


### Japanese

http://tenure5.vbl.okayama-u.ac.jp/HM_blog/?s=コドン
コドン | 酵母とシステムバイオロジー
- 2015-03-20 コドンの最適度はmRNAの安定性を決める主要な要因である。
- 2011-08-31 合成遺伝子で「コドンの最適化」をどう行なうのか？

2017/1/18
http://yfuruta.sakura.ne.jp/blog/?p=824
コドン最適化

2013年 第11号
https://www.sbj.or.jp/wp-content/uploads/file/sbj/9111/9111_biomedia_2(1).pdf
コドンが決め手!?組換えタンパク質の高発現法 今中 洋行

2010-08-25
http://d.hatena.ne.jp/morimori-08/20100825/1282745255
コドンバイアスはタンパク質のフォールディングのエラーを少なくする

https://www.nig.ac.jp/museum/evolution/04.html
遺伝暗号(コドン）使用の種による多様性

https://biowiki.edu-wiki.org/コドン表
コドン表 [BioWiki]

https://integbio.jp/dbcatalog/record/nbdc00033
Codon Usage Database - Integbio データベースカタログ

http://www.kazusa.or.jp/codon/
Codon Usage Database

http://www.genscript.jp/codon_opt.html
OptimumGene™コドン最適化技術 - GenScript

----------

----------


[コドン - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20130122)

[論文 - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20090605)

http://www.kri.sfc.keio.ac.jp/report/mori/2005/b-13.html
研究課題名        バイオインフォマティックスによる遺伝暗号解析

----------
### WCA
within-group correspondence analysis (WCA) of codon usage

Synonymous codon usage was investigated by within-group correspondence analysis (WCA) of codon counts [22, 23] and by correspondence analysis of relative synonymous codon usage (CA-RSCU). The latter is the most widely used method for codon usage analysis, whereas the former has been recently demonstrated to produce more unbiased results because it takes into account both amino acid composition and codon degeneracy information [48]. 


