----------

Haruo Suzuki (haruo[at]g-language[dot]org)  
Last Update: 2017-07-10

----------

# Codon usage

----------

## References

### Video

https://www.youtube.com/watch?v=gzYCZ80OygU
The Alien-Looking Codon Usages of Recently Acquired Genes Are Not Alien - Gary Olsen - YouTube
Published on Oct 7, 2015

https://www.youtube.com/watch?v=P-fjZPf3Dnw
Codon optimization: Why & how to design DNA sequences for optimal soluble protein expression - YouTube
Published on Nov 3, 2014

### Tools

http://codonw.sourceforge.net
Correspondence Analysis of Codon Usage
CodonW
15/Apr/2005

http://pbil.univ-lyon1.fr/datasets/charif04/
Online Synonymous Codon Usage Analyses with the ade4 and seqinR packages
2004

### Temperature

https://www.ncbi.nlm.nih.gov/pubmed/16989961
Gene. 2006 Dec 30;385:128-36. 
Synonymous codon usage and its potential link with optimal growth temperature in prokaryotes.
Lobry JR1, Necşulea A.

https://www.ncbi.nlm.nih.gov/pubmed/12364606
Nucleic Acids Res. 2002 Oct 1;30(19):4272-7.
Synonymous codon usage is subject to selection in thermophilic bacteria.
Lynn DJ1, Singer GA, Hickey DA.

### Amino acid usage

https://www.ncbi.nlm.nih.gov/pubmed/8065933
Nucleic Acids Res. 1994 Aug 11;22(15):3174-80.
Hydrophobicity, expressivity and aromaticity are the major trends of amino-acid usage in 999 Escherichia coli chromosome-encoded genes.
Lobry JR1, Gautier C.

### Review
http://www.ncbi.nlm.nih.gov/pubmed/21102527
Nat Rev Genet. 2011 Jan;12(1):32-42. doi: 10.1038/nrg2899. Epub 2010 Nov 23.
Synonymous but not the same: the causes and consequences of codon bias.
Plotkin JB1, Kudla G.

https://www.ncbi.nlm.nih.gov/pubmed/11719972
Curr Issues Mol Biol. 2001 Oct;3(4):91-7.
Synonymous codon usage in bacteria.
Ermolaeva MD1.


### Unclassified

https://www.ncbi.nlm.nih.gov/pubmed/10075995
Nucleic Acids Res. 1999 Apr 1;27(7):1642-9.
Proteome composition and codon usage in spirochaetes: species-specific and DNA strand-specific mutational biases.
Lafay B1, Lloyd AT, McLean MJ, Devine KM, Sharp PM, Wolfe KH.

https://www.ncbi.nlm.nih.gov/pubmed/15537809
Mol Biol Evol. 2005 Mar;22(3):547-61. Epub 2004 Nov 10.
Codon bias signatures, organization of microorganisms in codon space, and lifestyle.
Carbone A1, Képès F, Zinovyev A.

https://www.ncbi.nlm.nih.gov/pubmed/14594704
Bioinformatics. 2003 Nov 1;19(16):2005-15.
Codon adaptation index as a measure of dominating codon bias.
Carbone A1, Zinovyev A, Képès F.

### Plasmids
- [Davis JJ, Olsen GJ. Mol Biol Evol. 2010 Apr "Modal codon usage: assessing the typical codon usage of a genome."](https://www.ncbi.nlm.nih.gov/pubmed/20018979)
- [Guo FB, Yuan JB. DNA Res. 2009 Apr "Codon usages of genes on chromosome, and surprisingly, genes in plasmid are primarily affected by strand-specific mutational biases in Lawsonia intracellularis."](https://www.ncbi.nlm.nih.gov/pubmed/19221094)
- [Peixoto L et al. Gene. 2003 Nov 27 "The strength of translational selection for codon usage varies in the three replicons of Sinorhizobium meliloti."](https://www.ncbi.nlm.nih.gov/pubmed/14597394)

----------

[コドン - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20130122)

[論文 - Haruo Suzuki / Bioinformatics](http://d.hatena.ne.jp/haruosuz/20090605)

http://www.kri.sfc.keio.ac.jp/report/mori/2005/b-13.html
研究課題名        バイオインフォマティックスによる遺伝暗号解析

